### PR TITLE
Fix logging issue on windows

### LINKED ISSUE #5057: Logging is broken in production on Windows
## Summary

Logging to file was not working on Windows, as seen in #5030 -- seems likely to be multi-threaded concurrency issues (possibly `StreamHandler` being closed by one thread while still being used by another thread).

A temporary workaround to avoid bigger problems was introduced by https://github.com/learningequality/ka-lite/pull/5056, but it simply disables the `StreamHandler` output on Windows, thereby disabling logging.

We should remove the workaround introduced in #5056, and then sort out the original problem to get logging working properly again.
## Branch or installer method
- Branch: 0.16.x

### PR DESCRIPTION
**Pull request by mrpau-richard**
## Summary

Hi @benjaoming & @radinamatic  This fixes #5057

@radinamatic  Here's the [windows installer](https://drive.google.com/open?id=0B5Tyi-tRKOFMU3ppRGl4RmR6QTg) for testing the KA Lite windows logging.

### CODE REVIEW

**Comment by radinamatic** [comment]
@mrpau-richard Fix confirmed! 👍 

What a beautiful sight: logging on Windows again! 😄 

virtualbox_windows 7_16_12_2016_14_37_14

**Comment by benjaoming** [comment]
@radinamatic do you have a `server.log` with these contents, too? Does `--foreground` option still work, too?

**Comment by benjaoming** [comment]
This change seems to suppress final command line output.

Before:

```
➜  ka-lite git:(develop) bin/kalite start
Running 'kalite start' as daemon (system service)

Stand by while the server loads its data...

To access KA Lite from another connected computer, try the following address(es):
	http://192.168.1.100:8008/
	http://10.20.20.1:8008/
To access KA Lite from this machine, try the following address:
	http://127.0.0.1:8008/

Going to daemon mode, logging to /home/ben/.kalite/server.log
```

Now:

```
➜  ka-lite git:(mrpau-logging) bin/kalite start 
Running 'kalite start' as daemon (system service)

Stand by while the server loads its data...

Going to daemon mode, logging to /home/ben/.kalite/server.log
```

**Comment by mrpau-richard** [comment]
Hi @benjaoming done fixing the final command line output.

screen shot 2016-12-19 at 7 11 28 pm

Hi @radinamatic Here's the [windows installer](https://drive.google.com/open?id=0B5Tyi-tRKOFMSklEeTZKNHJVazg) for your testing.

**Comment by radinamatic** [comment]
@benjaoming Yes, `server.log` file is present, and `--foreground` option works!

`server.log` does not contain all the stuff that appears in the command prompt window, though...

virtualbox_windows 7_19_12_2016_15_45_15

**Comment by radinamatic** [comment]
On Windows this is the output:

```
C:\Users\IEUser>kalite start --foreground
Running 'kalite start' in foreground...

Stand by while the server loads its data...

To access KA Lite from another connected computer, try the following address(es)
:
        http://10.0.2.15:8008/
To access KA Lite from this machine, try the following address:
        http://127.0.0.1:8008/

[INFO] [2016-12-19 07:03:13,808] cherrypy.error: [19/Dec/2016:07:03:13] ENGINE L
istening for SIGTERM.
[INFO] [2016-12-19 07:03:13,812] cherrypy.error: [19/Dec/2016:07:03:13] ENGINE B
us STARTING
[INFO] [2016-12-19 07:03:13,815] cherrypy.error: [19/Dec/2016:07:03:13]  Loading
 and serving the Django application
[INFO] [2016-12-19 07:03:13,825] cherrypy.error: [19/Dec/2016:07:03:13] ENGINE S
tarted monitor thread '_TimeoutMonitor'.
[INFO] [2016-12-19 07:03:14,059] cherrypy.error: [19/Dec/2016:07:03:14] ENGINE S
erving on http://0.0.0.0:8008
[INFO] [2016-12-19 07:03:14,062] cherrypy.error: [19/Dec/2016:07:03:14] ENGINE B
us STARTED

```

**Comment by benjaoming** [comment]
@radinamatic 
Command-line output in your comparison starts at `06:35`, while the `server.log` starts `06:33`, so it's probably log data from previous invocation.

@mrpau-richard I'll test this later tonight, will merge when it's confirmed working :) Great fix!!

File: kalite/cli.py [hunk 1]
```diff
@@ -511,6 +509,9 @@ def start(debug=False, daemonize=True, args=[], skip_job_scheduler=False, port=N
 
     manage('initialize_kalite')
 
+    for addr in get_urls_proxy(output_pipe=sys.stdout):
+        sys.stdout.write("\t{}\n".format(addr))
+
```
Comment (benjaoming) [inline kalite/cli.py]: It seems you've moved the stuff that outputs to `sys.stdout`, yet kept the `print()` stuff above. Any reason for that?

I don't think there's a good reason why one of them is `sys.stdout`, while others are `print()`, but I can investigate this later...

**Comment by benjaoming** [comment]
@mrpau-richard - this is actually part of the PR template since merging another PR (which isn't part of yours, that's why tests are failing) - you should always add something in the release notes about your fixes or features. I'll add it after, so don't worry. Just letting you know :) ..it's to keep the `develop` branch "always releasable".

**Comment by benjaoming** [comment]
Confirming that stdout is working now again on Ubuntu, merging and making a bit of cleanup after.

**Pull request merged**